### PR TITLE
[css-contain] Apply different changes to improve tests

### PR DESCRIPTION
**Pull request by mrego**
Gérard Talbot shared some suggestions to improve css-contain tests suite:
http://www.gtalbot.org/BrowserBugsSection/CSS3Contain/CSS3Contain-ReviewedTests.txt

This patch applies the changes suggested in section:
APPROVED by me with some MODIFICATIONS, IMPROVEMENTS.

### CODE REVIEW

File: css/css-contain/contain-size-breaks-001.html [hunk 1]
```diff
@@ -31,18 +30,36 @@
...
-<div>A A A A A</div>
+<div>A B C D E</div>
 </article>
 
 <!--
-If the element is monolythic, it may:
-* overflow and be a 1x5 em rectangle
-* be sliced and be 2 contigious 1x2.5 em rectangles, appearing as a single
-  2x2.5 em rectangle.
+As the element is monolithic, it would be sliced in two contiguous 2.5em rectangles.
```
Comment (frivoal) [inline css/css-contain/contain-size-breaks-001.html]: The ASCII Art is good, but the comment isn't quite right. should be

> As the element is monolithic, it should either:
> * Overflow vertically, appearing as a signle 1x5 em rectangle
>  [... diagram...]
> * Be sliced, with one slice in each column, appearing as  a single 2x2.5 em rectangle
>  [... diagram...]

In other words, in the case where it overflows, it isn't being sliced.

File: css/css-contain/contain-layout-001.html [hunk 2]
```diff
@@ -11,9 +11,14 @@
 <style>
 span {
   contain: layout;
+  display: inline-block;
   height: 100vh; /*If layout containment applies, the span becomes a BFC, height applies, and knocks SS off the page */
 }
+
+#ss {
+  vertical-align: bottom;
```
Comment (frivoal) [inline css/css-contain/contain-layout-001.html]: I disagree with Gérard that this is necessary, as the initial value is baseline, which should be sufficient. 
That said, `vertical-align: bottom` isn't wrong, and maybe it is more explicit, so OK.

File: css/css-contain/contain-layout-001.html [hunk 3]
```diff
@@ -11,9 +11,14 @@
 <style>
 span {
   contain: layout;
+  display: inline-block;
```
Comment (frivoal) [inline css/css-contain/contain-layout-001.html]: Delete the `display: inline-block;` line. we're trying to test the absence of effect on non-atomic inlines, and this makes it into an atomic inline. Maybe an accidental leftover from debugging?

File: css/css-contain/contain-layout-001.html [hunk 4]
```diff
@@ -11,9 +11,14 @@
 <style>
 span {
```
Comment (frivoal) [inline css/css-contain/contain-layout-001.html]: This selector should not be `span`, but should only select the first of the two spans. Whether you want to add an id or use `:first-of-type` or whatever is up to you, but it should only select the first span.

**Review by frivoal — CHANGES_REQUESTED** [review]
Overall, it's an improvement (tests simplified, comments improved), but there are a couple of mistakes. Please fix first.

File: css/css-contain/contain-size-breaks-001.html (hunk 1 above)
Comment (mrego) [reply css/css-contain/contain-size-breaks-001.html]: Ok thanks for the clarification, I've updated the comment.

File: css/css-contain/contain-layout-001.html (hunk 2 above)
Comment (mrego) [reply css/css-contain/contain-layout-001.html]: Ok.

File: css/css-contain/contain-layout-001.html (hunk 3 above)
Comment (mrego) [reply css/css-contain/contain-layout-001.html]: Ooops, yeah sorry I was checking some things there and I forgot to delete it later.

File: css/css-contain/contain-layout-001.html (hunk 4 above)
Comment (mrego) [reply css/css-contain/contain-layout-001.html]: Done.

**Comment by mrego** [comment]
Thanks for the review, uploaded new version fixing the mistakes.

**Review by frivoal — APPROVED** [review]

**Pull request merged**